prow/cmd/deck/static/api/tide-history.ts

```diff
@@ -0,0 +1,23 @@
+
+export interface HistoryData {
+    History: {[key: string]: Record[]};
+}
+
+export interface Record {
+	time:    string;
+	action:  string;
+	baseSHA: string;
+	target:  PRMeta[];
+	err:     string;
+
+  // The following are not initially present and are instead populated based on the 'History' map key while filtering.
+  repo: string;
```

Comment:
(GitHub managed to discard this comment, so here I go again...)

The approach I've taken with these is to make the interfaces in `api/` match exactly the contents of the JSON provided by the backend. I'd prefer to keep that approach. Instead, I would create a new interface that extends this one in `tide-history/tide-history.ts`:

```ts
interface FilteredRecord extends Record {
  // The following are not initially present and are instead populated based on the 'History' map key while filtering.
  repo: string;
  branch: string;
}
```

Then in `redraw()`, on line 185 you can change `filteredRecs` to be a `FilteredRecord[]`, and construct it using new records by changing the construction of `rec` on line 200 to just this (and removing the following assignments):

```ts
const rec: FilteredRecord = {repo, branch, ...recs[i]};
```

You'll also have to change some function signatures (by my count, that's `redrawRecords` and `targetCell`) and remove some `Record` or `Record[]` type annotations (which I flagged as nits anyway because they are unnecessary).

Reply:
I switched to the pattern you recommended. 👍 

Related question: some of the fields are annotated as `omitempty` fields in the go code. Should the corresponding fields in the TS api files use `?` or will the fields receive the empty value if the JSON payload doesn't specify the field.

Reply:
The corresponding fields should use `?`; they'll be omitted if the JSON payload doesn't specify the field.

Effectively the `declare const tideHistory: TideHistory` is an implicit unsafe cast; if it turns out to _not_ conform to the promised type you will find out at runtime.

Reply:
Ok, fixed.